Update branching guide for 2.10 (#4021)
astronomer/docs: learn/airflow-branch-operator.md

If you have downstream tasks that need to run regardless of which branch is taken, like the `join` task in the previous example, you need to update the [trigger rule](airflow-trigger-rules.md). The default trigger rule in Airflow is `all_success`, which means that if upstream tasks are skipped, then the downstream task will not run. In the previous example, `none_failed_min_one_success` is specified to indicate that the task should run as long as one upstream task succeeded and no tasks failed.

> Starting with version 2.10, you can pass TaskGroups in some operators. If you provide a `task_group_id` to the operator instead of a `task_id`, it will be expanded into the roots of that task group. The operators that support this approach are:

Reviewer: ```suggestion
Starting with version 2.10, you can pass set a [task group](task-groups.md) as the direct downstream element of a branching task by returning its `task_group_id` in your decorated function or `python_callable` instead of a `task_id`. All root tasks of the task group run if the branching tasks returns the `task_group_id`.
```

Changed this slightly to make it work for both the traditional operators and the decorators :)

> - BaseBranchOperator

Reviewer: ```suggestion
```

I think we usually don't mention Base-level operators since DAG authors don't really interact with them, unless they write their own operators. But we could start 🤔 
Can leave this comment open for @kentdanas :)

- BranchPythonOperator
- BranchPythonVirtualenvOperator
- BranchExternalPythonOperator
- BranchDateTimeOperator
- LatestOnlyOperator
- BranchDayOfWeekOperator.

> 

Reviewer: I just tested with @task.branch and that works as well... looking at the PR I think all branching operators will support this change, so maybe we don't need a list here? 🤔 But if we have a list we would need to include the decorator versions as well.

Author: Thanks - I went ahead and removed the list but can add it back if needed.

If you have downstream tasks that need to run regardless of which branch is taken, like the `join` task in the previous example, you need to update the [trigger rule](airflow-trigger-rules.md). The default trigger rule in Airflow is `all_success`, which means that if upstream tasks are skipped, then the downstream task will not run. In the previous example, `none_failed_min_one_success` is specified to indicate that the task should run as long as one upstream task succeeded and no tasks failed.

> Starting with version 2.10, you can set a [task group](task-groups.md) as the direct downstream element of a branching task by returning its `task_group_id` in your decorated function or `python_callable` instead of a `task_id`. All root tasks of the task group run if the branching tasks returns the `task_group_id`.

Reviewer 2: ```suggestion
Starting with version 2.10, you can set a [task group](task-groups.md) as the direct downstream element of a branching task by returning its `task_group_id` in your decorated function or `python_callable` instead of a `task_id`. All root tasks of the task group run if the branching task returns the `task_group_id`.
```

Just to confirm, by "root" task we mean everything that doesn't have an upstream dependency within the task group?

Author: @TJaniF what do you think?

Author: @TJaniF went with just the DAG and a screenshot for simplicity. WDYT?
